subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/EclipseModel.java

```diff
@@ -55,25 +60,41 @@
  */
 public class EclipseModel {
 
-    private EclipseProject project;
+    private final Property<EclipseProject> project;
 
     private EclipseClasspath classpath;
 
-    private EclipseJdt jdt;
+    private final Property<EclipseJdt> jdt;
 
     private EclipseWtp wtp;
 
+    @Inject
+    public EclipseModel(ObjectFactory objectFactory) {
+        this.project = objectFactory.property(EclipseProject.class);
+        this.jdt = objectFactory.property(EclipseJdt.class);
+    }
+
     /**
      * Configures eclipse project information
      * <p>
      * For examples see docs for {@link EclipseProject}
      */
     public EclipseProject getProject() {
-        return project;
+        return project.getOrNull();
     }
 
     public void setProject(EclipseProject project) {
-        this.project = project;
+        this.project.set(project);
+    }
+
+    /**
+     * See {@link #setProject(EclipseProject)}.
+     *
+     * @since 4.9
+     */
+    @Incubating
+    public void setProject(Provider<EclipseProject> project) {
```

Comment:
This model object should own the EclipseProject object and take care of constructing it.